### PR TITLE
Second task 

### PR DESCRIPTION
**Pull request by bepo00**

### CODE REVIEW

File: app/src/main/java/com/frankie/app/ui/main/ui/main/SurveyListItemView.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    init {
+        setPadding(context.dpToPx(16))
+    }
+
+    fun bind(surveyData: SurveyData,
+             onSyncClicked: (SurveyData) -> Unit,
+             onPlayClicked: (SurveyData) -> Unit,
+             onInfoClicked: (SurveyData) -> Unit) {
+        binding.name.text = surveyData.name
+        binding.play.isEnabled = !surveyData.newVersionAvailable
```
Comment (askoura) [inline app/src/main/java/com/frankie/app/ui/main/ui/main/SurveyListItemView.kt:30]: we want an enabled/disabled clear state for both refresh and play.

play button should be enabled if no new version AND at least one design is downloaded and completed

File: app/src/main/java/com/frankie/app/business/survey/SurveyRepository.kt [hunk 2]
```diff
@@ -1,33 +1,151 @@
...
+                           private val surveyDataDao: SurveyDataDao,
+                           private val permissionDao: PermissionDao,
+                           private val sessionManager: SessionManager) : SurveyRepository {
+    override fun getSurveyList(): Flow<Result<List<SurveyData>>> {
         return flow {
-            emit(Result.success(service.getSurveyList()))
+            val surveyList = service.getSurveyList().map { survey ->
+                val design = service.getSurveyDesign(survey.id, PublishInfo())
+                val savedSurvey = surveyDataDao.getSurveyDataById(survey.id)
+                val newVersionAvailable = savedSurvey?.publishInfoEntity?.toPublishInfo()?.let { it != design.publishInfo }
```
Comment (askoura) [inline app/src/main/java/com/frankie/app/business/survey/SurveyRepository.kt:44]: This is always false for the first time a user logs in the app

File: app/src/main/java/com/frankie/app/api/survey/SurveyDesign.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import com.frankie.app.api.JsonNodeToStringSerializer
+import com.frankie.app.api.StringToJsonNodeDeserializer
+
+data class SurveyDesign(
+        @JsonProperty("files") val files: List<FileData>,
+        @JsonProperty("publishInfo") val publishInfo: PublishInfo,
+        @JsonProperty("validationJsonOutput")
+        @JsonSerialize(using = JsonNodeToStringSerializer::class)
+        @JsonDeserialize(using = StringToJsonNodeDeserializer::class)
+        val validationJsonOutput: String
```
Comment (askoura) [inline app/src/main/java/com/frankie/app/api/survey/SurveyDesign.kt:15]: I think it is better if you define validationJsonOutput as ObjectNode,  (this is the equivalent of JSONObject in Jackson) and then you don't need serialiser or deserialiser

File: app/src/main/java/com/frankie/app/business/auth/LogoutUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.frankie.app.business.auth
+
+import com.frankie.app.business.survey.SessionManager
+
+interface LogoutUseCase {
+    operator fun invoke()
```
Comment (askoura) [inline app/src/main/java/com/frankie/app/business/auth/LogoutUseCase.kt:6]: The backend has a logout method... let's also call it

File: app/src/main/java/com/frankie/app/ui/main/ui/main/MainViewModel.kt [hunk 5]
```diff
@@ -2,22 +2,44 @@ package com.frankie.app.ui.main.ui.main
...
+                    _state.update { _state.value.copy(isLoading = false, surveyList = result.getOrThrow()) }
+                } else {
+                    _state.update { _state.value.copy(isLoading = false) }
+                    processError(result.exceptionOrNull()!!)
+                }
+            }
+        }
+    }
+
+    fun surveyClicked(surveyData: SurveyData) {
```
Comment (askoura) [inline app/src/main/java/com/frankie/app/ui/main/ui/main/MainViewModel.kt:35]: let's rename to downloadDesign or something clearer

**Comment by askoura** [comment]
Also if it is not too much, please implement downloading x of y files

File: app/src/main/java/com/frankie/app/business/survey/SurveyRepository.kt (hunk 2 above)
Comment (bepo00) [reply app/src/main/java/com/frankie/app/business/survey/SurveyRepository.kt:43]: what do you mean? this method is called later also

Comment (askoura) [reply app/src/main/java/com/frankie/app/business/survey/SurveyRepository.kt:43]: let's take this on a call...

**Pull request merged**